zeroconf/__init__.py

```diff
@@ -1136,6 +1136,11 @@ def remove(self, entry: DNSRecord) -> None:
         try:
             list_ = self.cache[entry.key]
             list_.remove(entry)
+            # If we remove the last entry in the list
+            # we remove the key from the dict in order
+            # to avoid leaking memory
+            if not list_:
+                del self.cache[entry.key]
```

Comment:
While being cool to have this addressed, this belongs in a separate pull request, it's not a `handle_response()` optimization.

Reply:
Will make a new PR for this after this one is sorted out